src/main/java/com/hazelcast/remotecontroller/HazelcastCloudManager.java

```diff
@@ -0,0 +1,338 @@
+            String query = String.format("{\"query\": \"query { cluster(clusterId: \\\"%s\\\") { id name hazelcastVersion isTlsEnabled state discoveryTokens {source,token}}}\"}", clusterId);
+            String responseBody = createRequest(query).body().string();
+            LOG.info(maskValueOfToken(responseBody));
+            JsonNode rootNode = mapper.readTree(responseBody).get("data").get("cluster");
+            if(rootNode.asText().equalsIgnoreCase("null"))
+                return null;
+
+            CloudCluster cluster = new CloudCluster(rootNode.get("id").asText(), rootNode.get("name").asText(), getConnectionName(rootNode.get("id").asText()), rootNode.get("hazelcastVersion").asText(), rootNode.get("isTlsEnabled").asBoolean(), rootNode.get("state").asText(), rootNode.get("discoveryTokens").elements().next().get("token").asText(), null, null);
+            if(rootNode.get("isTlsEnabled").asBoolean())
+            {
+                cluster.setCertificatePath(downloadCertificatesAndGetPath(cluster.getId()));
+                cluster.setTlsPassword(getTlsPassword(cluster.getId()));
+            }
+            return cluster;
+        } catch (Exception e) {
+            e.printStackTrace();
+            return null;
+        }
+    }
+
+    public CloudCluster stopHazelcastCloudCluster(String clusterId) {
+        try {
+            String query = String.format("{\"query\": \"mutation { stopCluster(clusterId:\\\"%s\\\") { clusterId }}\"}", clusterId);
+            String responseBody = createRequest(query).body().string();
+            LOG.info(maskValueOfToken(responseBody));
+            if(waitForStateOfCluster(clusterId, "STOPPED", TimeUnit.MINUTES.toMillis(5)))
+                return getHazelcastCloudCluster(clusterId);
+            else
+                throw new Exception("State cannot come to STOPPED");
+        } catch (Exception e) {
+            e.printStackTrace();
+            return null;
+        }
+    }
+
+    public CloudCluster resumeHazelcastCloudCluster(String clusterId) {
+        try {
+            String query = String.format("{\"query\": \"mutation { resumeCluster(clusterId:\\\"%s\\\") { clusterId }}\"}", clusterId);
+            String responseBody = createRequest(query).body().string();
+            LOG.info(maskValueOfToken(responseBody));
+            if(waitForStateOfCluster(clusterId, "RUNNING", TimeUnit.MINUTES.toMillis(5)))
+                return getHazelcastCloudCluster(clusterId);
+            else
+                throw new Exception("State cannot come to RUNNING");
+        } catch (Exception e) {
+            e.printStackTrace();
+            return null;
+        }
+    }
+
+    public boolean deleteHazelcastCloudCluster(String clusterId) {
+        try {
+            String query = String.format("{\"query\": \"mutation { deleteCluster(clusterId:\\\"%s\\\") { clusterId }}\"}", clusterId);
+            String responseBody = createRequest(query).body().string();
+            LOG.info(responseBody);
```

Comment:
We are sure this does not include the token right?

Reply:
Right, if you would like to see token you need to ask for token also